incubator/buzzfeed-sso/templates/auth-deployment.yaml

```diff
@@ -0,0 +1,144 @@
+    chart: {{ template "buzzfeed-sso.chart" . }}
+    component: {{ $name }}-auth
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.auth.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ $name }}
+      component: {{ $name }}-auth
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+      labels:
+        app: {{ $name }}
+        component: {{ $name }}-auth
+        release: {{ .Release.Name }}
+    spec:
+    {{- if .Values.provider.google }}
+      volumes:
+        - name: google-service-account
+          secret:
+            secretName: {{ $googleSecret }}
+            items:
+            - key: service-account
+              path: sso-serviceaccount.json
+    {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}-auth
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command: ["/bin/sso-auth"]
+          ports:
+            - name: http
+              containerPort: 4180
+              protocol: TCP
+          env:
+            - name: SSO_EMAIL_DOMAIN
+              value: {{ .Values.emailDomain | quote }}
+            - name: HOST
+              value: {{ $authDomain }}
+            - name: REDIRECT_URL
+              value: https://{{ $authDomain }}
+            - name: PROXY_ROOT_DOMAIN
+              value: {{ .Values.rootDomain | quote }}
+            - name: PROXY_CLIENT_ID
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $authSecret }}
+                  key: proxy-client-id
+            - name: PROXY_CLIENT_SECRET
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $authSecret }}
+                  key: proxy-client-secret
+            - name: AUTH_CODE_SECRET
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $authSecret }}
+                  key: auth-code-secret
+            - name: COOKIE_SECRET
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $authSecret }}
+                  key: auth-cookie-secret
+            # # OLD_COOKIE_SECRET is the same as COOKIE_SECRET, not sure why its even needed at this point
+            - name: OLD_COOKIE_SECRET
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $authSecret }}
+                  key: auth-cookie-secret
+            # STATSD_HOST and STATSD_PORT must be defined or the app wont launch, they dont need to be a real host / port
+            - name: STATSD_HOST
+              value: localhost
+            - name: STATSD_PORT
+              value: "11111"
+            - name: COOKIE_SECURE
+              value: "true"
+            - name: CLUSTER
+              value: dev
+            - name: VIRTUAL_HOST
```

Comment:
@darioblanco I believe this environment variable is only useful when using [nginx-proxy](https://github.com/jwilder/nginx-proxy) as in the docker-compose example https://github.com/buzzfeed/sso/blob/master/quickstart/docker-compose.yml

Reply:
In any case, I would keep it as it is. We set the `VIRTUAL_HOST` to `auth.domain`, which is also used for the ingress configuration and the `PROVIDER_URL` in the proxy.
Unless we know 100% it is useless for Kubernetes deployments (for instance, I use the `nginx-ingress` and I haven't tested if it works without it).

Reply:
It works, I am removing this as requested, so we can reduce complexity